test/sequential/test-heapdump.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const common = require('../common');
+
+if (!common.isMainThread)
+  common.skip('process.chdir is not available in Workers');
+
+const { writeHeapSnapshot, getHeapSnapshot } = require('v8');
+const assert = require('assert');
+const fs = require('fs');
+const tmpdir = require('../common/tmpdir');
+
+tmpdir.refresh();
+process.chdir(tmpdir.path);
+
+{
+  writeHeapSnapshot('my.heapdump');
+  fs.accessSync('my.heapdump');
+}
+
+{
+  const heapdump = writeHeapSnapshot();
+  assert.strictEqual(typeof heapdump, 'string');
+  fs.accessSync(heapdump);
+}
+
+[1, true, {}, [], null, Infinity, NaN].forEach((i) => {
+  common.expectsError(() => writeHeapSnapshot(i), {
+    code: 'ERR_INVALID_ARG_TYPE',
+    type: TypeError,
+    message: 'The "path" argument must be one of type string, Buffer, or URL.' +
+             ` Received type ${typeof i}`
+  });
+});
+
+{
+  let data = '';
+  const snapshot = getHeapSnapshot();
+  snapshot.on('data', common.mustCallAtLeast((chunk) => {
+    data += chunk.toString();
```

Comment:
I suppose this can't result in `data` containing something that `JSON.parse()` won't accept but I'd still use `snapshot.setEncoding('utf8')`.